src/ILVerify/src/Program.cs

```diff
@@ -287,18 +287,17 @@ private void VerifyAssembly(PEReader peReader, EcmaModule module, string path)
         /// This method exists to avoid additional assembly resolving, which might be triggered by calling 
         /// MethodDesc.ToString().
         /// </summary>
-        private string GetQualifiedMethodName(MetadataReader metadataReader, MethodDefinitionHandle methodHandle)
+        private string GetQualifiedMethodName(string moduleName, MetadataReader metadataReader, MethodDefinitionHandle methodHandle)
         {
             var methodDef = metadataReader.GetMethodDefinition(methodHandle);
             var typeDef = metadataReader.GetTypeDefinition(methodDef.GetDeclaringType());
 
             var methodName = metadataReader.GetString(metadataReader.GetMethodDefinition(methodHandle).Name);
             var typeName = metadataReader.GetString(typeDef.Name);
             var namespaceName = metadataReader.GetString(typeDef.Namespace);
-            var assemblyName = metadataReader.GetString(metadataReader.GetAssemblyDefinition().Name);
```

Comment:
If this is the only line that is problematic, I would fix this by calling `metadataReader.IsAssembly` to check whether this is an assembly and if it isn't, call `metadataReader.GetModuleDefinition()` and fetch the name from that. Using an `AssemblyName`'s name as a module name is odd.

Note that the support for non-manifest modules in the type system is very small. If multi-module assemblies are a scenario you would want to support, we'll likely need to extend the type system to work with this better.

Reply:
Thanks I will do that.